### ISSUE TITLE
Squiz.CSS.SemicolonSpacingSniff false positive when style name proceeded by an asterisk

### ISSUE THREAD

**Issue by josephzidell** [squizlabs/PHP_CodeSniffer#2453]
The latest update of PHPCS fails on this code:
```css
.some_class {
	cursor: pointer;
	*cursor: hand;
}
```

With this message: `Style definitions must end with a semicolon`

**Comment by gsherwood** [comment]
Looks to have been caused by #2248

**Comment by josephzidell** [comment]
cc @jrfnl

**Comment by jrfnl** [comment]
@josephzidell I'd seen it. Pinging me is counter-productive.

**Comment by jrfnl** [comment]
@josephzidell Ok, looked into this. Fixing it is easy, that's not the problem, but I figured I should check what other prefixes the `T_STYLE` element could have to prevent similar issues cropping up later.

Well, I've searched through the CSS specs and while I can find plenty references to `*` as the universal selector and it's use in selectors in general, I can not for the life of me find any reference to the use of it at the start of, or within a _style name_.

Got any links to specifications for me ?

**Comment by josephzidell** [comment]
It's used in [datatables](https://github.com/DataTables/DataTables/blob/1_9/media/css/jquery.dataTables.css#L156). It's a backwards browser compatibility thing.

Here's a [stackoverflow](https://stackoverflow.com/questions/1690642/purpose-of-asterisk-before-a-css-property)

It seems the asterisk would be the only character

**Comment by josephzidell** [comment]
On a related note, [tailwind](https://tailwindcss.com/docs/configuration/#separator) uses a `:` separator for special class names (it's actually configurable)

**Comment by gsherwood** [comment]
@jrfnl Sorry, didn't mean for you to get called into this. I had a late meeting and ran out of time while looking into a fix.

**Comment by josephzidell** [comment]
My bad for pulling him into this.

**Comment by jrfnl** [comment]
> My bad for pulling him into this.

@josephzidell **_her_** thank you very much.

> @jrfnl Sorry, didn't mean for you to get called into this. I had a late meeting and ran out of time while looking into a fix.

@gsherwood :+1: I saw your response yesterday and wanted to avoid us both working on the same fix, which is why I left it at the time.

PR #2454 should fix this now anyhow.

**Comment by gsherwood** [comment]
> PR #2454 should fix this now anyhow.

Thanks a lot. Will get time to merge later today.

**Comment by josephzidell** [comment]
Thanks @jrfnl for the quick attention and fix!
